priam/src/main/java/com/netflix/priam/notification/BackupNotificationMgr.java

```diff
@@ -102,6 +107,42 @@ private Map<String, MessageAttributeValue> getMessageAttributes(
                 new MessageAttributeValue()
                         .withDataType("String")
                         .withStringValue(backupFileType.name()));
+
+        String additionalAttrs =
+                this.backupRestoreConfig.getBackupNotificationAdditionalMessageAttrs();
+        if (!StringUtils.isBlank(additionalAttrs)) {
+            try {
+                List<String> attrs =
+                        Splitter.on(",")
+                                .omitEmptyStrings()
+                                .trimResults()
+                                .splitToList(additionalAttrs);
+                for (String attr : attrs) {
```

Comment:
Prefer modern Java unless you have performance reasons to do otherwise.  So that means:

backupRestoreConfig.getBackupNotificationAdditionalMessageAttrs()
.stream()
.filter(attr -> message.has(attr))
.forEach(attr -> messageAttributes.put(attr, toStringAttribute(message.get(attr));

(or the version of that that actually compiles)

Reply:
I prefer idiomatic Java as well wherever I can but when there are checked exceptions thrown from the code being called, it does not make it look pretty. There is no performance benefit that I know of for using for loop instead of forEach in this case.